include/revng/Model/Helpers.h

```diff
@@ -186,6 +186,81 @@ inline std::string customName(const model::RawFunctionType &Function,
   return customName(Function, Argument.key());
 }
 
+template<EntityWithKey T, typename Root = model::Binary>
+inline std::string comment(const T &Value) {
+  static_assert(EntityWithComment<T>);
+  return detail::pathImpl<Root, T>(Value, "Comment");
```

Comment:
1. 👌.

> Why not simply accept `EntityWithComment`

`EntityWithComment` is not guaranteed to be an `EntityWithKey`, and this needs both to be true. I prefer this approach (one check for the major trait + one static assert for a more minor one) to a `requires` clause checking both concepts, but I'm not opposed to switching - if that's what you prefer.

> return `Value.Comment`

I have trouble understanding what you mean here. This function returns the path to the comment field for an arbitrary element of the binary (if it has one), how is returning value of said field related to anything?

Reply:
2.2 Mh, that was not clear at all.

Why are we concatenating strings instead of return a `TupleTreePath` and then let the user serialize it?

This seems a bit misguided.

Reply:
Because of ease of implementation. I thought about doing it through `TupleTreePath`s, but that started to seem like an overkill really fast (since it's one more level of indirection between the information we see and what we want to end up with), but if you'd prefer that, I can take a deeper look.

Reply:
Mmmh, not now. But if we concat strings, this will break when we change the serialization format.

Reply:
Consider that before I touched it, it was also serializing strings. So, at the very least, it's not  worse after compared to what it was before.

Reply:
OK, please leave a TODO about this and let's move on.

Reply:
Done.